lib/utils/join-sql-fragments.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+function doesNotWantLeadingSpace(str) {
+  return /^[;,)]/.test(str);
+}
+function doesNotWantTrailingSpace(str) {
+  return /\($/.test(str);
+}
+
+/**
+ * Joins an array of strings with a single space between them,
+ * except for:
+ * 
+ * - Strings starting with ';', ',' and ')', which do not get a leading space.
+ * - Strings ending with '(', which do not get a trailing space.
+ * 
+ * @param {string[]} parts
+ * @returns {string}
+ * @private
+ */
+function singleSpaceJoinHelper(parts) {
+  let result = '';
+  let skipNextLeadingSpace = true;
+  for (const part of parts) {
```

Comment:
This can be simplified using `parts.reduce`

Reply:
Hmm, to be honest I don't see a way to use `parts.reduce` here in a way that could benefit readability.

The only way to use reduce here that comes to my mind is like this:

```js
let skipNextLeadingSpace = true;
return parts.reduce((acc, part) => {
  let result;
  if (skipNextLeadingSpace || doesNotWantLeadingSpace(part)) {
    result = acc + part.trim();
  } else {
    result = acc + ` ${part.trim()}`;
  }
  skipNextLeadingSpace = doesNotWantTrailingSpace(part);
  return result;
}, '');
```

Do you have another suggestion? :)

Reply:
You can include both `skipNextLeadingSpace` and `result` in accumulator (as object). Then do  `parts.reduce(() => {}).result` without using `skipNextLeadingSpace` in outer closure

Reply:
@sushantdhiman Hmmm, to be honest I don't think that improves readability, but I did it anyway :)